utils/tensorboard_parser.py

```diff
@@ -0,0 +1,143 @@
+# Standard Library
+import argparse
+import json
+
+
+class threadId:
+    def __init__(self, tid, thread_name):
+        self.tid = tid
+        self.thread_name = thread_name
+
+
+class processId:
+    def __init__(self, id, name):
+        self.id = id
+        self.name = name
+        self._threads = dict()
+
+    def add_thread(self, threadid, thread_name):
+        self._threads[threadid] = threadId(threadid, thread_name)
+
+    def get_thread_info(self, threadid):
+        return self._threads[threadid]
+
+
+class XEvent:
+    def __init__(self, ts, name, dur, pid, tid, event_args):
+        self.start_time = ts
+        self.event_name = name
+        self.duration = dur
+        self.end_time = self.start_time + self.duration
+        self.pid = pid
+        self.tid = tid
+        self.event_args = event_args
+
+
+class tensorboardEvents:
+    def __init__(self, trace_file):
+        self.trace_json_file = trace_file
+        self.timescale = "ns"
+        self._processes = dict()
+        self._x_events = list()
+        self._start_time_stamp = 0
+        self._start_time_known = False
+
+        try:
+            with open(trace_json_file) as json_data:
+                trace_json_data = json.load(json_data)
+        except Exception as e:
+            print(f"Error No metric file for {trace_json_file}: Exception {str(e)} ")
+            status = False
+            return
+
+        trace_events = trace_json_data["traceEvents"]
+        self.timescale = trace_json_data["displayTimeUnit"]
```

Comment:
these should be conditional only if key is present in json file. 
There can 3 different types of json file - 
One which we emit from c++ framework
Other which we emit from python side
3rd one emitted by horovod
4th one emitted by tensorboard. 

May be in constructor you can take type.